### PR TITLE
fix: show bounding box for 3 or more linear point elements

### PR DESCRIPTION
**Pull request by ad1992**
for https://github.com/excalidraw/excalidraw/issues/5535

https://user-images.githubusercontent.com/11256141/183853657-d3c561d0-d0df-482e-ad6f-08ef28fac4a7.mp4

### CODE REVIEW

**Comment by dwelle** [comment]
I'd decrease the bounding box offset from the line. For that matter, we may not do any special offset at first and gather feedback first. Also note that in a future PR we may wanna remove N/S/W/E handles as discussed previously, so it'll make it even less likely the points will conflict with the bounding box (except the start/end point which will be common, true).

**Comment by dwelle** [comment]
Ok, so after further discussion, it seems we're in a bind and it's unclear how to resolve it:

1. making the bounding box align with the element's actual bounding box makes the points often collide with the resize handles which is not a great UX (and also doesn't look good)

	image

2. offsetting the bounding box has its own host of issues:

	- doesn't look good either. The rendered bounding box is inconsistent to all the other elements. The expectation is broken.

		(in red what the actual bounding box should be)

		image
	- the resizing experience is weird. You go to the negative while your cursor is still inside the element:  
	
		excal-linear-bounding-box-discrepancy

	- grid aligning is affected / unintuitive
	
		image

File: src/element/transformHandles.ts [hunk 1]
```diff
@@ -248,12 +252,31 @@ export const getTransformHandles = (
...
+  const element = elements[0];
+  if (!isLinearElement(element)) {
+    return true;
+  }
+
+  const points = LinearElementEditor.getPointsGlobalCoordinates(element);
+  if (points.length > 2) {
+    return true;
+  }
+  return false;
```
Comment (dwelle) [inline src/element/transformHandles.ts:281]: why not just `return element.points.length > 2` ?

File: src/element/transformHandles.ts [hunk 2]
```diff
@@ -248,12 +252,31 @@ export const getTransformHandles = (
   } else if (isTextElement(element)) {
     omitSides = OMIT_SIDES_FOR_TEXT_ELEMENT;
   }
-
+  const dashedLineMargin = isLinearElement(element) ? 12 : 4;
```
Comment (dwelle) [inline src/element/transformHandles.ts:255]: is the `12` in any way related to the `8` we use in rendering? Also, I know we'll be removing it soon, but I'd use constants for these so we can later easily find all occurrences to replace.

Comment (ad1992) [reply src/element/transformHandles.ts:254]: I guess you are referring to `renderSelectionBorder` ? Nope I don't think so. 
Both have their own calculations so should be fine.

**Review by dwelle — APPROVED** [review]

**Pull request merged**